pytorch_lightning/loggers/base.py

```diff
@@ -338,6 +338,19 @@ def _add_prefix(self, metrics: Dict[str, float]):
 
         return metrics
 
+    def on_save_checkpoint(self, trainer, pl_module: LightningModule, checkpoint: Dict[str, Any]) -> dict:
+        """
+        Called when saving a model checkpoint, use to persist state of logger.
+
+        Args:
+            trainer: the current Trainer instance.
+            pl_module: the current LightningModule instance.
+            checkpoint: the checkpoint dictionary that will be saved.
+
+        Returns:
+            The logger state.
+        """
+
```

Comment:
I think this shouldn't be `@abstractmethod` because not all loggers should be forced to define it right?